File: Week3/machine-learning-ex2/ex2/plotData.m
function plotData(X, y)
%PLOTDATA Plots the data points X and y into a new figure 
%   PLOTDATA(x,y) plots the data points with + for the positive examples
%   and o for the negative examples. X is assumed to be a Mx2 matrix.

% Create New Figure
figure; hold on;

% ====================== YOUR CODE HERE ======================
% Instructions: Plot the positive and negative examples on a
%               2D plot, using the option 'k+' for the positive
%               examples and 'ko' for the negative examples.
%

pos = X(y == 1, :);
neg = X(y == 0, :);

scatter(pos(:, 1), pos(:, 2), 'k+', 'LineWidth', 2)
scatter(neg(:, 1), neg(:, 2), 'o', 'LineWidth', 1, ...
    'MarkerFaceColor', 'y', 'MarkerEdgeColor', 'k')

% =========================================================================

hold off;

end

File: Week2/machine-learning-ex1/ex1/Python/plotData.py
import matplotlib.pyplot as plt


def plot_data(x, y):

    # plot_data Plots the data points x and y into a new figure
    # plot_data(x,y) plots the data points and gives the figure axes labels of
    # population and profit.
    #
    #  ====================== YOUR CODE HERE ======================
    #  Instructions: Plot the training data into a figure using the
    #                "figure" and "plot" commands. Set the axes labels using
    #                the "xlabel" and "ylabel" commands. Assume the
    #                population and revenue data have been passed in
    #                as the x and y arguments of this function.
    #
    #  Hint: You can use the 'rx' option with plot to have the markers
    #        appear as red crosses. Furthermore, you can make the
    #        markers larger by using plot(..., 'rx', 'MarkerSize', 10);

    plt.plot(x, y, 'rx')   # Plot the data
    plt.ylabel('Profit in $10,000s')   # Set the y-axis
    plt.xlabel('Population of City in 10,000s')    # Set the x-axis

